core/src/test/scala/tech/beshu/ror/unit/configuration/EsConfigFileLoaderTest.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.unit.configuration
+
+import better.files.File
+import io.circe.Decoder
+import org.scalatest.Matchers._
+import org.scalatest.WordSpec
+import tech.beshu.ror.configuration.{EsConfigFileLoader, MalformedSettings}
+import tech.beshu.ror.providers.EnvVarsProvider
+
+class EsConfigFileLoaderTest
+  extends WordSpec {
+  private implicit val envVarsProvider: EnvVarsProvider = name =>
+    name.value.value match {
+      case "USER_NAME" => Some("John")
+      case _ => None
+    }
+
+  "FileConfigLoader" should {
+    "decode file file" in {
+      val result = loadFromTempFile[String](""""encoded value"""")
+      result.right.get shouldBe "encoded value"
```

Comment:
IMO it should be checked as follows:
`result shouldBe Right("encoded value")`

error message produced by scalatest will be better